Katana/Store.swift

```diff
@@ -18,10 +18,22 @@ public protocol AnyStore: class {
    Dispatches a `Dispatchable` item
    
    - parameter dispatchable: the dispatchable to dispatch
-   - returns: a promise that is resolved when the dispatchable is handled by the store
-  */
+   - returns: a promise parameterized to void that is resolved when the dispatchable is handled by the store
+   */
   @discardableResult
-  func dispatch(_ dispatchable: Dispatchable) -> Promise<Void>
+  func dispatch<T: Dispatchable>(_ dispatchable: T) -> Promise<Void>
+  
+  /**
+   Dispatches a `SideEffect` item
+   
+   - parameter dispatchable: the Side Effect to dispatch
+   - returns: a promise parameterized to the side effect's return value, that is resolved when the dispatchable is handled by the store
+   */
+  @discardableResult
+  func dispatch<T: SideEffect>(_ dispatchable: T) -> Promise<T.ReturnValue>
```

Comment:
Why don't we have the StateUpdater equivalent that returns `Promise<Void>`?

Reply:
see above